pyuvm/reg/uvm_reg.py

```diff
@@ -0,0 +1,935 @@
+        regfile_parent: uvm_reg_file = None,
+        hdl_path: str = "",
+        throw_error_on_read: bool = False,
+        throw_error_on_write: bool = False,
+        **kwargs,
+    ) -> None:
+        # TODO: remove backward compatibility
+        if (
+            not isinstance(regfile_parent, uvm_reg_file) and regfile_parent is not None
+        ) or "addr" in kwargs:
+            warnings.warn(
+                "The 'configure' definition as changed, the address should be "
+                "specified when the register is added to a map"
+                "'blk_parent' instead",
+                DeprecationWarning,
+                2,
+            )
+            for arg, val in kwargs.items():
+                if arg == "addr":
+                    self._addr = val
+                else:
+                    raise SyntaxError(f"Unknown argument {arg}")
+            if (
+                not isinstance(regfile_parent, uvm_reg_file)
+                or regfile_parent is not None
+            ):
+                self._addr = regfile_parent
+        # END: backward compatibility
+
+        if blk_parent is None:
+            logger.error(
+                "uvm_reg.configure() called without a parent block "
+                f"for instance {self.get_name()} of register type "
+                f"{self.get_type_name()}."
+            )
+            return
+        self._parent = blk_parent
+        self._parent._add_register(self)
+        # TODO: remove backward compatibility
+        if self._addr is not None:
+            try:
+                self.build()
+            except AttributeError as e:
+                logger.error(
+                    "Calling build() when not implemented by user"
+                    "This is not the way to go, this is only for "
+                    "backward compatibility, please follow the UVM "
+                    "guidelines"
+                )
+                raise e
+        else:
+            # END: backward compatibility
+            self._regfile_parent = regfile_parent
+        if hdl_path != "":
+            self.add_hdl_path_slice(hdl_path, -1, -1)
+        # TODO: remove backward compatibility
+        try:
+            if throw_error_on_read is None or throw_error_on_read is not None:
+                pass
+        except NameError:
+            warnings.warn("The 'throw_error_on_read' argument is deprecated")
+        try:
+            if throw_error_on_write is None or throw_error_on_write is not None:
+                pass
+        except NameError:
+            warnings.warn("The 'throw_error_on_write' argument is deprecated")
+        # END: backward compatibility
+
+    def set_offset(
+        self, map: uvm_reg_map, offset: uvm_reg_addr_t, unmapped: bool = False
+    ) -> None:
+        if len(self._maps) > 1 and not map:
+            logger.error(
+                f"'set_offset' requires a map when register "
+                f"'{self.get_full_name()}' belongs to more than one map."
```

Comment:
```suggestion
                f"{self.get_full_name()!r} belongs to more than one map."
```
Using `repr()` instead of the default `str()`.